### PR TITLE
Remove indirect mozjs dependency from compositing

### LINKED ISSUE #19939: Can't build servo_url without mozjs
I'm trying to use the `compositing` crate without having to include `mozjs`. My problem is that one of its dependencies, `servo_url`, doesn't compile unless `malloc_size_of` has the `servo` feature enabled. Enabling the feature creates a dependency on `mozjs`.

It seems a bit excessive for `servo_url` to depend on `mozjs`, so I'm hoping it's not intentional?

Here's the compilation error:

```
  --> [...]/.cargo/git/checkouts/servo-12bfca331f128de3/1662fd7/components/url/origin.rs:14:41
   |
14 | #[derive(Clone, Debug, Deserialize, Eq, MallocSizeOf, PartialEq, Serialize)]
   |                                         ^^^^^^^^^^^^ the trait `malloc_size_of::MallocSizeOf` is not implemented for `url::Host`
```

The root cause is that `malloc_size_of` doesn't implement traits for URLs without the `servo` feature.

### PR DESCRIPTION
**Pull request by anniryynanen**
- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #19939.

- [ ] There are tests for these changes OR
- [X] These changes do not require tests because there is no change to functionality.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19948)

### CODE REVIEW

File: components/malloc_size_of/Cargo.toml [hunk 1]
```diff
@@ -9,7 +9,9 @@ publish = false
 path = "lib.rs"
 
 [features]
-servo = ["mozjs", "string_cache", "url", "webrender_api", "xml5ever"]
+servo = ["mozjs", "string_cache", "xml5ever"]
+url_traits = ["url"]
+webrender_traits = ["webrender_api"]
```
Comment (SimonSapin) [inline components/malloc_size_of/Cargo.toml]: Adding two features that enable one crate each is probably unnecessary. Optional dependencies can already be used as features: `#[cfg(feature = "url")]`, `features = ["url"]`

Comment (anniryynanen) [reply components/malloc_size_of/Cargo.toml]: I didn't know you could do that, it certainly makes things a lot cleaner. :) I'll make the changes tomorrow.

**Review by jdm — APPROVED** [review]

**Comment by anniryynanen** [comment]
I rebased the commit and it should merge cleanly now, but the `S-needs-rebase` label remains. Is there something else I should do?

**Comment by jdm** [comment]
@bors-servo r+
Nope. That's just a longstanding bug with our github integration robot.

**Pull request merged**